nifi-nar-bundles/nifi-salesforce-bundle/nifi-salesforce-processors/src/main/java/org/apache/nifi/processors/salesforce/util/SalesforceRestService.java

```diff
@@ -103,6 +103,10 @@ public InputStream postRecord(String sObjectApiName, String body) {
         return request(request);
     }
 
+    public String getVersionedBaseUrl() {
+        return baseUrl + "/services/data/v" + version;
+    }
+
     private InputStream request(Request request) {
```

Comment:
I recommend refactoring this method a little bit.

`private InputStream request(Request request) {
        try (Response response = httpClient.newCall(request).execute()) {
            if (response.code() < 200 || response.code() > 201) {
                throw new ProcessException("Invalid response" +
                        " Code: " + response.code() +
                        " Message: " + response.message() +
                        " Body: " + (response.body() == null ? null : response.body().string())
                );
            }
            return Objects.requireNonNull(response.body()).byteStream();
        } catch(ProcessException e) {
            throw e;
        } catch (Exception e) {
            throw new ProcessException(String.format("Salesforce HTTP request failed [%s]", request.url()), e);
        }
    }`